src/main/java/Main.java

```diff
@@ -165,6 +172,14 @@ public static void main(String[] args) throws Exception {
         }
     }
 
+    /**
+     * Returns true if the Java runtime version check should not be done, and any version allowed.
+     * @see #ENABLE_FUTURE_JAVA_CLI_SWITCH
+     */
+    private static boolean isFutureJavaEnabled(String[] args) {
+        return hasArgument(ENABLE_FUTURE_JAVA_CLI_SWITCH, args) || Boolean.parseBoolean(System.getenv("ENABLE_FUTURE_JAVA"));
```

Comment:
Just for curiosity: why aren't we using prefixes for environment variables to avoid collision with existing ones? i.e: _JENKINS_ENABLE_FUTURE_JAVA_ or something like that.

Reply:
done